src/shared/functions.py

```diff
@@ -182,6 +182,22 @@ def move_s3_file(s3_client, s3_bucket_name: str, old_key: str, new_key: str) ->
         raise S3UploadError
 
 
+def get_s3_keys(s3_client, s3_bucket_name: str, prefix: str, token: str | None = None) -> list:
+    """Gets the list of all keys in S3 starting with the prefix"""
+    if token:
+        res = s3_client.list_objects_v2(
+            Bucket=s3_bucket_name, Prefix=prefix, ContinuationToken=token
+        )
+    else:
+        res = s3_client.list_objects_v2(Bucket=s3_bucket_name, Prefix=prefix)
```

Comment:
nit: couldn't you just unconditionally pass along `ContinuationToken=token`? I assume that `list_objects_v2` has a default of `None` there, such that these two paths are equal? Did not read docs though :smile:

Reply:
I tried that, and no, unfortunately.